src/data-interfaces/HoloFuelDnaInterface.js

```diff
@@ -12,8 +12,20 @@ const createZomeCall = instanceCreateZomeCall(INSTANCE_ID)
 
 const MOCK_DEADLINE = '4019-01-02T03:04:05.678901234+00:00'
 
+/* Creates an array of all transactions that inlcude new counterparties for a provided transaction list and counterparty list */
+export const findnewCounterpartiesFromList = (transactionList = [], counterpartyList = []) => {
+  // eslint-disable-next-line array-callback-return
+  const newCouterpartyList = transactionList.filter(({ counterparty }) => {
+    if (isEmpty(counterpartyList)) return counterparty
+    const existingCounterparty = counterpartyList.find(counterpartyInList => counterpartyInList.id === counterparty.id)
+    if (!existingCounterparty) return counterparty
+  })
+  const uniqueNewCounterpartyTransactions = _.uniqBy(newCouterpartyList, 'counterparty')
```

Comment:
I'm not sure about this, but I don't think this actually gets you uniqueness. the counterparty field of each transaction object will be a different object, even if counterparty.id is the same.

Anyway, I think I would make uniqueness the responsibility of `getTxCounterparties`. Seems like there's never any value in that function returning duplicate counterparties.

Reply:
Looking at `getTxCounterparties`, it looks like it already de-dupes the list, so no need to do it here.